### PR TITLE
fix(stoneintg-358|stonebugs-42):drop URL & report number of vulns

### PR DESCRIPTION
**Pull request by jsztuka**
example output after the change:

```
[jsztuka@fedora hacbs-test]$ conftest test --no-fail ~/Documents/playground/tkn/clair-result.json --policy policies/clair/vulnerabilities-check.rego --namespace required_checks --output=json
[
        {
                "filename": "/home/jsztuka/Documents/playground/tkn/clair-result.json",
                "namespace": "required_checks",
                "successes": 1,
                "failures": [
                        {
                                "msg": "Found packages with high vulnerabilities. Consider updating to newer version of those packages, they may not longet be affected by the reported CVEs. You can check clair logs for more infromation. Packages found: expat, krb5-libs, libksba, python39, python39-libs, systemd, systemd-libs, systemd-pam, urllib3, wheel, xz-libs",
                                "metadata": {
                                        "details": {
                                                "description": "The image shouldn't contain packages that have high severity vulnerabilities.",
                                                "name": "clair_high_vulnerabilities"
                                        },
                                        "vulnerabilities-count": 11
                                }
                        },
                        {
                                "msg": "Found packages with medium vulnerabilities. Consider updating to newer version of those packages, they may not longet be affected by the reported CVEs. You can check clair logs for more infromation. Packages found: containers-common, criu, curl, dbus, dbus-common, dbus-daemon, dbus-libs, dbus-tools, expat, fuse-overlayfs, gnupg2, gnutls, libcom_err, libcurl, libgcrypt, libslirp, libtasn1, libxml2, openssl-libs, pcre2, platform-python, platform-python-setuptools, python3-libs, python3-libxml2, python3-setuptools-wheel, python39, python39-libs, runc, setuptools, skopeo, slirp4netns, sqlite-libs, systemd, systemd-libs, systemd-pam, tar, vim-minimal, zlib",
                                "metadata": {
                                        "details": {
                                                "description": "The image shouldn't contain packages that have medium severity vulnerabilities.",
                                                "name": "clair_medium_vulnerabilities"
                                        },
                                        "vulnerabilities-count": 38
                                }
                        },
                        {
                                "msg": "Found packages with low vulnerabilities. Consider updating to newer version of those packages, they may not longet be affected by the reported CVEs. You can check clair logs for more infromation. Packages found: containers-common, criu, fuse-overlayfs, libslirp, runc, skopeo, slirp4netns",
                                "metadata": {
                                        "details": {
                                                "description": "The image shouldn't contain packages that have low severity vulnerabilities.",
                                                "name": "clair_low_vulnerabilities"
                                        },
                                        "vulnerabilities-count": 7
                                }
                        }
                ]
        }
]
```

### CODE REVIEW

File: policies/clair/vulnerabilities-check.rego [hunk 1]
```diff
@@ -1,42 +1,41 @@
...
 
-violation_critical_vulnerabilities[{"msg": msg, "details":{"name": name, "description": description, "url": url}}] {
+violation_critical_vulnerabilities[{"msg": msg, "vulnerabilities-count": vulns_count, "details":{"name": name, "description": description}}] {
   rpms_with_critical_vulnerabilities := {rpm.Name | rpm := input.data[_].Features[_]; count(rpm.Vulnerabilities) > 0; rpm.Vulnerabilities[_].Severity == "Critical"}
   not count(rpms_with_critical_vulnerabilities) == 0
 
   name := "clair_critical_vulnerabilities"
-  msg = sprintf("Found packages with critical vulnerabilities: %s", [concat(", ", rpms_with_critical_vulnerabilities)])
-  description := "The image musn't contain packages that have critical severity vulnerabilities."
-  url := "https://source.redhat.com/groups/public/product-security/content/product_security_folder/understanding_red_hat_security_vulnerabilities_10pdf"
```
Comment (dirgim) [inline policies/clair/vulnerabilities-check.rego:10]: Removing the `url` field will be in conflict with the [API contract](https://github.com/redhat-appstudio/book/blob/main/ADR/0013-integration-service-api-contracts.md#detailed-conftest-output-json) for this format of output. If we don't have a good alternative at this time, we can maybe have it as empty for now?

Comment (jsztuka) [reply policies/clair/vulnerabilities-check.rego:10]: Great catch, yep, lets keep it empty for now.

File: policies/clair/vulnerabilities-check.rego [hunk 2]
```diff
@@ -1,42 +1,45 @@
...
+violation_critical_vulnerabilities[{"msg": msg, "vulnerabilities-count": vulns_count, "details":{"name": name, "description": description, "url": url}}] {
   rpms_with_critical_vulnerabilities := {rpm.Name | rpm := input.data[_].Features[_]; count(rpm.Vulnerabilities) > 0; rpm.Vulnerabilities[_].Severity == "Critical"}
   not count(rpms_with_critical_vulnerabilities) == 0
 
   name := "clair_critical_vulnerabilities"
-  msg = sprintf("Found packages with critical vulnerabilities: %s", [concat(", ", rpms_with_critical_vulnerabilities)])
-  description := "The image musn't contain packages that have critical severity vulnerabilities."
-  url := "https://source.redhat.com/groups/public/product-security/content/product_security_folder/understanding_red_hat_security_vulnerabilities_10pdf"
+  vulns_count = count(rpms_with_critical_vulnerabilities)
+  msg = sprintf("Found packages with critical vulnerabilities. Consider updating to newer version of those packages, they may not longet be affected by the reported CVEs. You can check clair logs for more infromation. Packages found: %s", [concat(", ", rpms_with_critical_vulnerabilities)])
```
Comment (mattreid) [inline policies/clair/vulnerabilities-check.rego:9]: typo - no longer

File: policies/clair/vulnerabilities-check.rego [hunk 3]
```diff
@@ -1,42 +1,45 @@
...
+  description := "The image shouldn't contain packages that have critical severity vulnerabilities."
```
Comment (mattreid) [inline policies/clair/vulnerabilities-check.rego:10]: How will msg and description work together? Would it make sense to split the information where msg states that a category check found affected packages, with the recommendation to update them, and the packages are listed as the description? 

"The image shouldn't contain packages that have <category> severity vulnerabilities" doesn't seem that useful to me if we have the new msg text.

Comment (jsztuka) [reply policies/clair/vulnerabilities-check.rego:10]: Sounds about right, MSG in current state would be too overwhelming.
Agreed on: MSG - states a category check found packages with recommendation
DESCRIPTION: list all the packages
If you agree I would replace the whole sentence in description with affected packages.

**Comment by jsztuka** [comment]
>Could we change how these are reported to be warnings, instead of failures? The enterprise contract policy (if applied) determines what is or is not acceptable.
The way that clair-action reports the output needs to remain.
That change would need to be applied in build-definitions repo, where the task definition is. I can create a follow up PR regarding that change.

**Comment by jsztuka** [comment]
Had to change naming convention from `vulnerabilities_count` to `vulnerabilities_number`. Because of unsafe naming
```
unittests/test_clair/vulnerabilities-check_test.rego:8: rego_unsafe_var_error: var _ is unsafe
unittests/test_clair/vulnerabilities-check_test.rego:8: rego_unsafe_var_error: var count is unsafe
```

**Comment by jsztuka** [comment]
@dirgim does the updated unit tests suffice?

File: policies/clair/vulnerabilities-check.rego (hunk 2 above)
Comment (jsztuka) [reply policies/clair/vulnerabilities-check.rego]: updated

File: policies/clair/vulnerabilities-check.rego [hunk 4]
```diff
@@ -1,42 +1,45 @@
...
+violation_critical_vulnerabilities[{"msg": msg, "vulnerabilities_number": vulns_num, "details":{"name": name, "description": description, "url": url}}] {
   rpms_with_critical_vulnerabilities := {rpm.Name | rpm := input.data[_].Features[_]; count(rpm.Vulnerabilities) > 0; rpm.Vulnerabilities[_].Severity == "Critical"}
   not count(rpms_with_critical_vulnerabilities) == 0
 
   name := "clair_critical_vulnerabilities"
-  msg = sprintf("Found packages with critical vulnerabilities: %s", [concat(", ", rpms_with_critical_vulnerabilities)])
-  description := "The image musn't contain packages that have critical severity vulnerabilities."
-  url := "https://source.redhat.com/groups/public/product-security/content/product_security_folder/understanding_red_hat_security_vulnerabilities_10pdf"
+  vulns_num = count(rpms_with_critical_vulnerabilities)
+  msg := "Found packages with critical vulnerabilities. Consider updating to a newer version of those packages, they may no longer be affected by the reported CVEs. You can check clair logs for more infromation."
```
Comment (mattreid) [inline policies/clair/vulnerabilities-check.rego:9]: There's a typo on "infromation", but personally, I don't see a lot of value in saying "You can check clair logs for more infromation." with each category message. They're viewing that log right now, they just need to scroll up to the previous step.

Comment (mattreid) [reply policies/clair/vulnerabilities-check.rego:9]: Everything else looks good to me @jsztuka, thanks for making the other changes!

Comment (jsztuka) [reply policies/clair/vulnerabilities-check.rego]: Included the update note change in previously mentioned story.

File: unittests/test_clair/vulnerabilities-check_test.rego
Comment (mattreid) [inline unittests/test_clair/vulnerabilities-check_test.rego:1]: Do these need to update to remove the last sentence in the result msg too, so it matches the above rego?

Comment (jsztuka) [reply unittests/test_clair/vulnerabilities-check_test.rego:1]: Correct, I was testing if unittests fail the check, forgot to push changes. Thanks for reminding!

**Review by dirgim — APPROVED** [review]
LGTM

**Review by mattreid — APPROVED** [review]
I'm not the best option to review code, but from what I can tell, LGTM :)

**Pull request merged**